### PR TITLE
feat(ironfish): Add burn method to wallet

### PR DESCRIPTION
**Pull request by rohanjadvani**
## Summary

Add a method to the wallet to burn an asset

## Testing Plan

Added unit tests

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/wallet/wallet.test.ts [hunk 1]
```diff
@@ -933,6 +933,109 @@ describe('Accounts', () => {
...
+      const { node } = await nodeTest.createSetup()
+      const account = await useAccountFixture(node.wallet)
+
+      const mined = await useMinerBlockFixture(node.chain, 2, account)
+      await expect(node.chain).toAddBlock(mined)
+      await node.wallet.updateHead()
+
+      const asset = new Asset(account.spendingKey, 'mint-asset', 'metadata')
+      const mintValue = BigInt(10)
+      const mintBlock = await useBlockFixture(node.chain, async () => {
```
Comment (NullSoldier) [inline ironfish/src/wallet/wallet.test.ts:947]: you could use my new useMintBlockFixture from my other open PR https://github.com/iron-fish/ironfish/pull/2810

Comment (rohanjadvani) [reply ironfish/src/wallet/wallet.test.ts]: Updated

File: ironfish/src/wallet/wallet.ts [hunk 2]
```diff
@@ -739,6 +739,49 @@ export class Wallet {
...
+    const heaviestHead = this.chain.head
+    if (heaviestHead === null) {
+      throw new Error('You must have a genesis block to create a transaction')
+    }
+
+    expiration = expiration ?? heaviestHead.sequence + transactionExpirationDelta
+    if (this.chain.verifier.isExpiredSequence(expiration, this.chain.head.sequence)) {
+      throw new Error('Invalid expiration sequence for transaction')
+    }
+
```
Comment (jowparks) [inline ironfish/src/wallet/wallet.ts:760]: Should we do any validation of assetIdentifier here, or is it done inside of `createTransaction`?

File: ironfish/src/wallet/wallet.ts [hunk 3]
```diff
@@ -739,6 +739,49 @@ export class Wallet {
...
+      expiration,
+    )
+
+    const transaction = await this.postTransaction(raw)
+
+    const verify = this.chain.verifier.verifyCreatedTransaction(transaction)
+    if (!verify.valid) {
+      throw new Error(`Invalid transaction, reason: ${String(verify.reason)}`)
+    }
+
```
Comment (jowparks) [inline ironfish/src/wallet/wallet.ts:776]: Someone who has been working on wallet should probably validate these are done in correct order/nothing is missing

File: ironfish/src/wallet/wallet.ts (hunk 2 above)
Comment (rohanjadvani) [reply ironfish/src/wallet/wallet.ts:760]: Creating and posting the transaction will validate the sender owns note(s) [+ correct witness path] with adequate balance to burn the provided value for the given asset identifier

**Comment by jowparks** [comment]
Going to approve to unblock, this is low risk given current state of chain

**Review by jowparks — APPROVED** [review]

**Review by hughy — APPROVED** [review]
LGTM 👍

**Review by NullSoldier — APPROVED** [review]

File: ironfish/src/wallet/wallet.ts (hunk 3 above)
Comment (NullSoldier) [reply ironfish/src/wallet/wallet.ts:776]: approved

**Pull request merged**